src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Tokenization.cs

```diff
@@ -342,5 +381,58 @@ private static TokenType MapNodeKindToTokenType(NodeKind kind)
                     return TokenType.Unknown;
             }
         }
+
+        /// <summary>
+        /// Adds token to the given list of tokens if it is feasible (token should not be skipped).
+        /// </summary>
+        /// <param name="tokens">List of tokens to which given token might be added.</param>
+        /// <param name="token">Token to be added if feasible.</param>
+        /// <param name="tokenTypesToSkip">Optional: Token types to be skipped.</param>
+        /// <param name="expression">Optional: Expression.</param>
+        private static void AddTokenIfPossible(this ICollection<ITokenTextSpan> tokens, ITokenTextSpan token, IReadOnlyCollection<TokenType> tokenTypesToSkip = null, string expression = null)
+        {
+            if (token != null && !ShouldBeSkipped(token, tokenTypesToSkip, expression))
+            {
+                tokens.Add(token);
+            }
+        }
+
+        /// <summary>
+        /// Determines where the given token type should be skipped or not.
+        /// </summary>
+        /// <param name="tokenType">Token type to be skipped.</param>
+        /// <param name="tokenTypesToSkip">Optional: Token types to be skipped.</param>
+        /// <returns>True if this token type should be skipped or false otherwise.</returns>
+        private static bool ShouldBeSkipped(this TokenType tokenType, IReadOnlyCollection<TokenType> tokenTypesToSkip = null)
+        {
+            return tokenTypesToSkip != null && tokenTypesToSkip.Contains(tokenType);
+        }
+
+        /// <summary>
+        /// Determines where the given token should be skipped or not.
+        /// </summary>
+        /// <param name="token">Token to be skipped.</param>
+        /// <param name="tokenTypesToSkip">Optional: Token types to be skipped.</param>
+        /// <param name="expression">Optional: Expression.</param>
+        /// <returns>True if this token should be skipped or false otherwise.</returns>
+        private static bool ShouldBeSkipped(ITokenTextSpan token, IReadOnlyCollection<TokenType> tokenTypesToSkip = null, string expression = null)
+        {
+            var shouldBeSkippedBasedOnTokenType = token.TokenType.ShouldBeSkipped(tokenTypesToSkip);
```

Comment:
>token.TokenType.ShouldBeSkipped(tokenTypesToSkip); [](http://example.com/codeflow?start=50&length=50)

Do we need to worry about nulls?token?.TokenType?.ShouldBeSkipped(tokenTypesToSkip);

Reply:
That's not possible at all here. We are the ones who create these token objects inside a "private" function so need to worry about that.